flink-core/src/main/java/org/apache/flink/api/common/state/StateTtlConfig.java

```diff
@@ -205,15 +247,26 @@ public Builder neverReturnExpired() {
 		 *
 		 * @param timeCharacteristic The time characteristic configures time scale to use for ttl.
 		 */
+		@Deprecated
```

Comment:
when we deprecate, we should also say what to use instead in doc comment, like `FoldingState`.